### PR TITLE
Add PHP 7.4 to Travis CI.

### PR DESCRIPTION
**Pull request by connorshea**
## Description
This is essentially the same as #7974, but built on hotfix-7.10.x and with the fix from #7976. I wanted to see if other issues crop up now that that one is fixed.

## Motivation and Context
PHP 7.4 is coming out at the end of November and we should try to be ready for that.

## How To Test This
See if CI passes.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by connorshea** [comment]
It has the same issue as #7974, but I don't know what causes it :/



(Also the PHP version check seems a bit buggy)

**Comment by connorshea** [comment]
I tested PHP 7.4 with an updated PHPUnit in #7988 and that didn't fix it, so I guess the version checker not recognizing 7.4.0-dev is the source of the problem?

**Comment by lazka** [comment]
Maybe because of these errors linked in the job log: https://www.simpleupload.co.uk/uploads/U9kQlPUserWizardCest.testScenarioInstallSuiteCRMWithDefaultConfiguration.fail.html.txt

**Comment by connorshea** [comment]
*shrug* I've got no idea why it's getting stuck like this.

**Comment by connorshea** [comment]
Okay, finally the error has changed. The main change that was necessary was having the version check just return `1` all the time.

Now the failure outputs this screenshot:

File: travis.php.ini [hunk 1]
```diff
@@ -2,4 +2,4 @@ memory_limit = -1
 display_errors = On
 log_errors = On
 trace_errors = On
-error_log = /var/log/apache2/error.log
```
Comment (Dillon-Brown) [inline travis.php.ini]: Nice! I had been meaning to look into why this wasn't posting for a while. Going to cherry-pick your improved logging commit and throw up a quick PR so we can get this merged before all the PHP 7.4 stuff.

**Comment by connorshea** [comment]
Okay, so apparently backporting #8073 actually fixed the install failing. Woo.

Now we can to the other deprecations, _yay_

**Comment by connorshea** [comment]
I think the current PHPUnit failure is caused by a fix from https://github.com/ezyang/htmlpurifier/pull/224, so we'll need to wait for a new release of HTMLPurifier before this'll work.

Unfortunately, the StateChecker freaks out and will probably cause everything else to fail until HTMLPurifier is updated :/

**Comment by connorshea** [comment]
Lots of failures like `Trying to access array offset on value of type null` or `Trying to access array offset on value of type boolean`.

**Comment by connorshea** [comment]
At this point, most of the remaining failures are essentially waiting on the state checker's removal and for HTMLPurifier to push out a new release.

**Comment by connorshea** [comment]
Based on the results from #8170, about half of the remaining errors in the unit tests come from the state checker, so I guess I'll wait for that to be fixed before I go any further on this.

**Comment by connorshea** [comment]
:/ Most of the remaining errors are caused by `$mod_strings` being `null`. I'm not really sure how to even fix that.

**Comment by connorshea** [comment]
The best way to fix this would be to figure out how it's even happening, I guess. Seems like these should always be loaded and I'm not sure why they're not.

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/SuiteCRM/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/SuiteCRM/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by connorshea** [comment]
PHP 7.4 isn't working on Travis yet because the php-build project doesn't quite have a working build yet: https://travis-ci.community/t/add-php-7-4-branch-to-test-against/2179/16

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/SuiteCRM/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/samus-aran/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/samus-aran/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/samus-aran/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/samus-aran/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by connorshea** [comment]
I have absolutely no clue why the CRM isn't letting me use 7.4.0. It doesn't make any sense.

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/samus-aran/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/samus-aran/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/samus-aran/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by connorshea** [comment]
```php
echo "PHP_VERSION is {$sys_php_version}. SUITECRM_PHP_MIN_VERSION is {$min_php_version}. SUITECRM_PHP_REC_VERSION is {$rec_php_version}.";

# output:
# PHP_VERSION is 7.4.0. SUITECRM_PHP_MIN_VERSION is 5.5.9. SUITECRM_PHP_REC_VERSION is 7.1.0.
```
??? Why does it think it's EOL then!??

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/samus-aran/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by samus-aran** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/samus-aran/SuiteCRM/pullRequest?prid=4280306&bid=14602262)

**Comment by connorshea** [comment]
Apparently the buggy version checker was buggy and broke with PHP 7.4. I'm... not entirely sure how.

Anyway, that problem is now fixed. Hurray.

**Comment by connorshea** [comment]
Now that that problem's resolved, I can get back to actually making the tests pass on 7.4.

I should probably just drop this PR, port over the deprecation fixes that still need to be fixed, and then let someone else (*cough* Dillon *cough*) figure out why the `$mod_strings` global variable is undefined sometimes. I'd rather not rebase this branch right now since it's such a mess.

**Comment by connorshea** [comment]
I'm closing this in favor of #8530.

**Pull request closed without merging**